test_util/aws.py

```diff
@@ -0,0 +1,109 @@
+import logging
+from collections import namedtuple
+
+import boto3
+
+LOGGING_FORMAT = '[%(asctime)s|%(name)s|%(levelname)s]: %(message)s'
+logging.basicConfig(format=LOGGING_FORMAT, level=logging.DEBUG)
+logger = logging.getLogger(__name__)
+
+Host = namedtuple('Host', ['private_ip', 'public_ip'])
+
+
+class BotoWrapper():
+    def __init__(self, region, aws_access_key_id, aws_secret_access_key):
+        self.region = region
+        self.session = boto3.session.Session(
+            aws_access_key_id=aws_access_key_id, aws_secret_access_key=aws_secret_access_key)
+
+    def client(self, name):
+        return self.session.client(service_name=name, region_name=self.region)
+
+    def resource(self, name):
+        return self.session.resource(service_name=name, region_name=self.region)
+
+    def create_key_pair(self, key_name):
+        """Retruns private key of newly generated pair
+        """
+        key = self.resource('ec2').KeyPair(key_name)
+        return key.key_material
+
+    def delete_key_pair(self, key_name):
+        self.resource('ec2').KeyPair(key_name).delete()
+
+    def create_stack(self, name, template_url, user_parameters, deploy_timeout=60):
+        """Returns boto stack object
+        """
+        logging.info('Requesting AWS CloudFormation...')
+        cf_parameters = []
+        for k, v in user_parameters.items():
+            cf_parameters.append({'ParameterKey': k, 'ParameterValue': v})
+        self.resource('cloudformation').create_stack(
+            StackName=name,
+            TemplateURL=template_url,
+            DisableRollback=True,
+            TimeoutInMinutes=deploy_timeout,
+            Capabilities=['CAPABILITY_IAM'],
+            Parameters=cf_parameters)
+        return CFStack(name, self)
+
+
+class CFStack():
+    def __init__(self, stack_name, boto_wrapper):
+        self.boto_wrapper = boto_wrapper
+        self.stack = self.boto_wrapper.resource('cloudformation').Stack(stack_name)
+
+    def wait_for_stack_creation(self):
+        self.boto_wrapper.client('cloudformation').get_waiter('stack_create_complete').\
+            wait(StackName=self.stack.stack_name)
+
+    def wait_for_stack_deletion(self):
+        self.boto_wrapper.client('cloudformation').get_waiter('stack_delete_complete').\
+            wait(StackName=self.stack.stack_name)
+
+
+class DcosCfSimple(CFStack):
+    @classmethod
+    def create(cls, stack_name, template_url, public_agents, private_agents,
+               admin_location, key_pair_name, boto_wrapper):
+        parameters = {
+            'KeyName': key_pair_name,
+            'AdminLocation': admin_location,
+            'PublicSlaveInstanceCount': str(public_agents),
+            'SlaveInstanceCount': str(private_agents)}
+        boto_wrapper.create_stack(stack_name, template_url, parameters)
+        return cls(stack_name, boto_wrapper)
+
+    def delete(self):
+        logger.info('Starting deletion of CF stack')
+        self.stack.delete()
```

Comment:
Sounds good, I was thinking we wanted to make sure it was all based on a stack id rather than a stack name since from the docs stack_id based you can access after deletion